examples/bert/run_pretraining.py

```diff
@@ -291,6 +301,38 @@ def train_step(self, data):
         self.nsp_accuracy.update_state(data["next_sentence_labels"], nsp_preds)
         return {m.name: m.result() for m in self.metrics}
 
+class LinearDecayWithWarmup(keras.optimizers.schedules.LearningRateSchedule):
+    """
+    Implements a learning rate schedule for BERT.
+
+    This schedule implements a linear warmup for the first `num_warmup_steps` 
+    and for the steps post that, till the `num_train_steps` it will implement 
+    a linear decay.
+    """
+    
+    def __init__(self,
+            learning_rate,
+            num_warmup_steps,
+            num_train_steps
+        ):
+        self.threshold_learning_rate = learning_rate
+        self.warmup_steps = num_warmup_steps
+        self.train_steps = num_train_steps
+    
+    def __call__(self, step):
+        final_lr = tf.cast(self.threshold_learning_rate, dtype = tf.float32)
```

Comment:
Maybe call this peak_lr? final_lr is actually 0.0.

Reply:
Yeah, makes sense. I was also thinking of changing it to threshold itself.